sklearn/cluster/tests/test_k_means.py

```diff
@@ -1030,3 +1030,47 @@ def test_minibatch_kmeans_wrong_params(param, match):
     # are passed for the MiniBatchKMeans specific parameters
     with pytest.raises(ValueError, match=match):
         MiniBatchKMeans(**param).fit(X)
+
+
+@pytest.mark.parametrize("param, match", [
+    ({"n_local_trials": 0},
+     r"n_local_trials is set to 0 but should be an "
+     r"integer value greater than zero"),
+    ({"x_squared_norms": X[:2]},
+     r"The length of x_squared_norms .* should "
+     r"be equal to the length of n_samples")]
+)
+def test_kmeans_plusplus_wrong_params(param, match):
+    with pytest.raises(ValueError, match=match):
+        kmeans_plusplus(X, n_clusters, **param)
+
+
+@pytest.mark.parametrize("data", [X, X_csr])
```

Comment:
Could you also parametrize the `dtype` with `[np.float64, np.float32]`? Just call data = data.astype(dtype) prior to calling `kmeans_plusplus`.